src/gui/advancedsettings.cpp

```diff
@@ -219,8 +224,15 @@ void AdvancedSettings::saveAdvancedSettings()
 #endif
     // Disk queue size
     session->setDiskQueueSize(m_spinBoxDiskQueueSize.value() * 1024);
+#if (LIBTORRENT_VERSION_NUM >= 20006)
+    // Disk IO read mode
+    session->setDiskIOReadMode(static_cast<BitTorrent::DiskIOReadMode>(m_comboBoxDiskIOReadMode.currentIndex()));
+    // Disk IO write mode
+    session->setDiskIOWriteMode(static_cast<BitTorrent::DiskIOWriteMode>(m_comboBoxDiskIOWriteMode.currentIndex()));
```

Comment:
Don't use `currentIndex` for mapping, see how `m_comboBoxOSMemoryPriority` do it properly.

Reply:
Currently I strongly believe that the way from #16895 is most reliable.